### PR TITLE
topology2: cleanup basic definitions

### PR DESCRIPTION
**Pull request by plbossart**
I don't know how we managed to have so many blatant logical inversions and inconsistencies.

dai_in/out
ibs/obs
aif_in/aif_out
capture/playback direction and names

and all of the above.

If it looks bad it's because it is. We need stronger class definitions to avoid such issues.

### CODE REVIEW

**Comment by marc-hb** [comment]
>  so many blatant logical inversions and inconsistencies.

Do we have no test coverage for these? If not, why and how were they submitted?

File: tools/topology/topology2/platform/intel/sdw-amp-generic.conf [hunk 1]
```diff
@@ -163,7 +163,7 @@ IncludeByKey.SDW_AMP_FEEDBACK {
 				}
 			]
 
-			passthrough-be [
+			passthrough-capture-be [
```
Comment (ranj063) [inline tools/topology/topology2/platform/intel/sdw-amp-generic.conf:164]: @plbossart this is not the wrong direction. The original passthrough-be is applicable for both directions but the passthrough-capture-be limits to 32-bit support only

File: tools/topology/topology2/include/pipelines/cavs/passthrough-capture.conf [hunk 2]
```diff
@@ -16,9 +16,9 @@
 # Where N is the unique pipeline ID within the same alsaconf node.
 #
 
-<include/common/audio_format.conf>
 <include/components/copier.conf>
 <include/components/pipeline.conf>
+<include/common/audio_format.conf>
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/passthrough-capture.conf:21]: is this an unrelated change? the original order is in alphabetical order no?

File: tools/topology/topology2/cavs-src-mixin-mixout-hda.conf [hunk 3]
```diff
@@ -73,7 +73,7 @@ Object.Pipeline {
 		}
 	]
 
-	passthrough-be [
+	passthrough-playback-be [
```
Comment (ranj063) [inline tools/topology/topology2/cavs-src-mixin-mixout-hda.conf]: this is wrong @plbossart. Like I mentioned passthrough-be is not only for playback. If you look at the direction below, it is meant for capture. So changing this to playback is wrong

File: tools/topology/topology2/include/pipelines/cavs/passthrough-playback-be.conf [hunk 4]
```diff
@@ -32,17 +32,12 @@ Class.Pipeline."passthrough-playback-be" {
...
-		# mixin-playback objects instantiated within the same alsaconf node must have
-		# unique pipeline_id attribute
-		#
 		unique	"instance"
 	}
 
 	Object.Widget {
 		copier."1" {
-			type dai_in
-			node_type $HDA_LINK_OUTPUT_CLASS
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/passthrough-playback-be.conf:45]: if you remove the default node_type, I think we should make sure it is set in all top-level conf files first. I think we avoid setting it for HDA

File: tools/topology/topology2/include/pipelines/cavs/deepbuffer-playback.conf [hunk 5]
```diff
@@ -49,6 +49,7 @@ Class.Pipeline."deepbuffer-playback" {
 			copier_type	"host"
 			type	"aif_in"
 			node_type $HDA_HOST_OUTPUT_CLASS
+
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/deepbuffer-playback.conf:52]: spurious change

File: tools/topology/topology2/include/pipelines/cavs/mixout-gain-dai-copier-playback.conf [hunk 6]
```diff
@@ -47,7 +47,7 @@ Class.Pipeline."mixout-gain-dai-copier-playback" {
 	Object.Widget {
 		mixout."1" {}
 		copier."1" {
-			type dai_in
+			type dai_out
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/mixout-gain-dai-copier-playback.conf:50]: I am not sure about this one @plbossart . I think you made too many changes in this PR that are not correct and you've got failures left, right & center. It might be a good idea to go step by step and separate the clean ups one by one

File: tools/topology/topology2/include/pipelines/cavs/passthrough-playback.conf [hunk 7]
```diff
@@ -62,7 +62,7 @@ Class.Pipeline."passthrough-playback" {
 				in_valid_bit_depth	24
 				out_bit_depth		32
 				out_valid_bit_depth	32
-				dma_buffer_size "$[$obs * 2]"
+				dma_buffer_size "$[$ibs * 2]"
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/passthrough-playback.conf:66]: and oh I forgot t mention, we need to remove dma_buffer_size altogether. This is no longer used by the kernel and it computed based on ibs/obs and stream direction

File: tools/topology/topology2/include/pipelines/cavs/passthrough-capture.conf (hunk 2 above)
Comment (plbossart) [reply tools/topology/topology2/include/pipelines/cavs/passthrough-capture.conf:21]: it's to align passthrough-capture and passthrough-playback. One of them doesn't follow the convention. I don't care what the convention is, I just want the two files to be aligned.

File: tools/topology/topology2/cavs-src-mixin-mixout-hda.conf (hunk 3 above)
Comment (plbossart) [reply tools/topology/topology2/cavs-src-mixin-mixout-hda.conf]: It's all needlessly complicated. No wonder no one can figure this out.

File: tools/topology/topology2/include/pipelines/cavs/passthrough-playback-be.conf (hunk 4 above)
Comment (plbossart) [reply tools/topology/topology2/include/pipelines/cavs/passthrough-playback-be.conf:45]: I don't know how to progress. This is wrong in general. It should never have been set this way, if there is an attribute that is not valid by default across all uses, then it cannot be initialized by default in the base class.

File: tools/topology/topology2/include/pipelines/cavs/deepbuffer-playback.conf (hunk 5 above)
Comment (plbossart) [reply tools/topology/topology2/include/pipelines/cavs/deepbuffer-playback.conf:52]: no, this is intentional. 

this was to make sure all uses of the copier_type, type and node_type are well isolated to make sure we can detect any differences.

File: tools/topology/topology2/platform/intel/sdw-amp-generic.conf (hunk 1 above)
Comment (plbossart) [reply tools/topology/topology2/platform/intel/sdw-amp-generic.conf:164]: And btw why is this named passthrough in the first place. This should be 'copier-be' or 'generic-be'. The passthrough property depends on how copiers are connected. A copier does not have in itself any passthrough attribute.

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by plbossart** [comment]
> Can one of the admins verify this patch?

broken CI @lgirdwood

**Comment by plbossart** [comment]
> @plbossart any chance you can split this PR up as it could be only 1 or 2 patches that are problematic. i.e. we can apply the fixes that are passing CI and revisiste the failure patches ?

I'll close this PR and let @ranj063 take over. I broke too many eggs with my changes.
